File: app/src/main/java/site/madcat/appnotes/UI/EditNoteFragment.java
package site.madcat.appnotes.UI;

import android.content.Context;
import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.EditText;

import java.io.Serializable;

import site.madcat.appnotes.R;
import site.madcat.appnotes.domain.Note;

public class EditNoteFragment extends Fragment {
    private EditText titleEditText;
    private EditText bodyEditText;
    private Button saveChangeButton;
    public Note item;
    private int id;
    private Controller controller;

    public EditNoteFragment() {}

    interface Controller {
        void replaceToListFragment(Note note);
        void replaceToListFragment();
    }

    @Override
    public void onAttach(@NonNull Context context) {
        super.onAttach(context);
        if (context instanceof Controller) {
            controller = (Controller) context;
        } else {
            //todo
        }
    }

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
    }


    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        return inflater.inflate(R.layout.fragment_edit_note, container, false);
    }


    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        initialsView();
        Bundle argument = getArguments();
        if (argument != null) {
            Note note = (Note) argument.getSerializable(Note.class.getSimpleName());
            titleEditText.setText(note.getTitle());
            bodyEditText.setText(note.getNoteBody());
            id = note.getId();
            getNoteFromList(item);
        }

        saveChangeButton.setOnClickListener(v -> {
            if (((ListActivity) requireActivity()).newRecord == true) {
                Note newNote = new Note(
                        titleEditText.getText().toString(),
                        bodyEditText.getText().toString()

                );
                controller.replaceToListFragment(newNote);

            } else {
                controller.replaceToListFragment();//
            }


        });
    }

    private void initialsView() {
        titleEditText = getActivity().findViewById(R.id.title_edit_text);
        bodyEditText = getActivity().findViewById(R.id.body_edit_text);
        saveChangeButton = getActivity().findViewById(R.id.save_change_button);
    }

    private void getNoteFromList(Note note) {
        if (note != null) {
            titleEditText.setText(note.getTitle());
            bodyEditText.setText(note.getNoteBody());
            id = note.getId();
        } else {

        }
    }

    public static EditNoteFragment setInputArgumentsNoteFrames(Note note) {
        EditNoteFragment editNoteFragment = new EditNoteFragment();
        Bundle bundle = new Bundle();
        bundle.putSerializable(Note.class.getSimpleName(), note);
        editNoteFragment.setArguments(bundle);
        return editNoteFragment;
    }


}

File: app/src/main/java/site/madcat/appnotes/UI/EditNoteFragment.kt
package site.madcat.appnotes.UI

import android.content.Context
import android.widget.EditText
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Button
import androidx.fragment.app.Fragment
import com.google.mlkit.vision.text.Text
import site.madcat.appnotes.R
import site.madcat.appnotes.UI.EditNoteFragment
import site.madcat.appnotes.domain.Note

class EditNoteFragment : Fragment() {
    private var controller: Controller? = null //для того чтоб убить в onDestroy
    private lateinit var titleEditText: EditText
    private lateinit var detailEditText: EditText
    private lateinit var saveChangeButton: Button

    internal var id = 0
    private var oldTitle: String? = null
    private var oldDetail: String? = null


    internal interface Controller {
        fun loadList()
        val screenOrientation: Boolean
        fun refreshAdapter()
        fun editNote(id: Int, title: String?, detail: String?)
        fun addNewNote(title: String?, detail: String?)
    }

    override fun onAttach(context: Context) {
        super.onAttach(context)
        if (context is Controller) {
            controller = context
        } else {
            //todo
        }
    }


    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        return inflater.inflate(R.layout.fragment_edit_note, container, false)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        initialsView(view)
        params
        saveChangeButton.setOnClickListener { v: View? ->
            if ((titleEditText.text.isNotEmpty()) or (detailEditText.text.isNotEmpty())) {

                if ((requireActivity() as ListActivity).newRecord == true) {
                    controller!!.addNewNote(
                        titleEditText.text.toString(),
                        detailEditText.text.toString()
                    )
                    (requireActivity() as ListActivity).newRecord = false
                    clearView()
                } else {
                    if ((oldTitle != titleEditText.text.toString()) or (oldTitle != titleEditText.text.toString())) {
                        controller!!.editNote(
                            id,
                            titleEditText.text.toString(),
                            detailEditText.text.toString()
                        )
                    }
                }
                if (controller!!.screenOrientation == true) {
                    controller!!.loadList()
                }
            }
        }
    }

    override fun onDestroy() {
        controller = null
        super.onDestroy()
    }

    val params: Unit
        get() {
            val argument = arguments
            if (argument != null) {
                val note = argument.getSerializable(Note::class.java.simpleName) as Note?
                (requireActivity() as ListActivity).newRecord = false
                getNoteFromList(note)
            } else {
                (requireActivity() as ListActivity).newRecord = true
            }
        }

    private fun clearView() {
        titleEditText.text = null
        detailEditText.text = null
    }

    private fun initialsView(view: View) {
        titleEditText = view.findViewById(R.id.title_edit_text)
        detailEditText = view.findViewById(R.id.body_edit_text)
        saveChangeButton = view.findViewById(R.id.save_change_button)

    }

    private fun getNoteFromList(note: Note?) {
        if (note != null) {
            oldTitle = note.title
            oldDetail = note.noteDetail
            titleEditText.setText(note.title)
            detailEditText.setText(note.noteDetail)
            id = note.id
        }
    }

    fun setInputArgumentsNoteFrames(note: Note?): EditNoteFragment {
        val editNoteFragment = EditNoteFragment()
        val bundle = Bundle()
        bundle.putSerializable(Note::class.java.simpleName, note)
        editNoteFragment.arguments = bundle
        return editNoteFragment
    }
}